### ISSUE TITLE
ON CONFLICT RETURNING incorrect data returned on v2.1.0-beta.20180924

### ISSUE THREAD

**Issue by nsf** [cockroachdb/cockroach#30786]
**BUG REPORT**

Bug is observed on docker image `cockroachdb/cockroach-unstable:v2.1.0-beta.20180924`

SQL:

```sql
DROP DATABASE IF EXISTS foobar;
CREATE DATABASE foobar;

SET database = foobar;

DROP TABLE IF EXISTS foobar;
CREATE TABLE foobar (
	id INT NOT NULL DEFAULT unique_rowid(),
	a int not null,
	b int not null,
	CONSTRAINT "primary" PRIMARY KEY (id ASC),
	UNIQUE INDEX foobar_uniq (a, b)
);

INSERT INTO foobar (a, b) values (1, 1), (1, 2) ON CONFLICT (a, b) DO NOTHING RETURNING id, a, b;
```

Result:
```
DROP DATABASE
CREATE DATABASE
SET
DROP TABLE
CREATE TABLE
id      a       b
1       1       387103546467024897
1       2       387103546467057665
```

As you can see CDB messes up the columns, reporting "id" on "b". Previously it worked fine, for example on `cockroachdb/cockroach:v2.0.5`:

```
DROP DATABASE
CREATE DATABASE
SET
DROP TABLE
CREATE TABLE
id      a       b
387103986403115009      1       1
387103986403147777      1       2
```

It is related to ON CONFLICT (a, b) DO NOTHING part of the SQL statement, works fine without it.

**Comment by jordanlewis** [comment]
Thanks for the report.

@BramGruneir can you take a look?

**Comment by BramGruneir** [comment]
Yep, I'll take a look.  I thought the rewrite of the rewriting logic fixed this.  Thanks for the report @nsf !

**Comment by BramGruneir** [comment]
I tried this use case on both our 2.1 branch and on master and this has been fixed since #30467.  The rewrite did indeed fix this.

@nsf Please let me know if you are still seeing this in our latest beta.
